ndc-client/src/apis/configuration.rs

```diff
@@ -1,11 +1,14 @@
+use reqwest::header::{HeaderMap, HeaderValue};
+
 #[derive(Debug, Clone)]
 pub struct Configuration {
     pub base_path: String,
     pub user_agent: Option<String>,
     pub client: reqwest::Client,
```

Comment:
`client` provides a way to edit the default set of headers. Why not use that?

Reply:
Are you talking about [`ClientBuilder`](https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html)?

So, the Configuration will have the client (already configured with default headers). While initializing, the user will have to make sure that the appropriate headers are already part of the client.

I am okay with this change, the only downside is that while initializing, we will have to create a [`ClientBuilder`](https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html) first, then [add default headers](https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.default_headers) and then create client (and handle errors as the [`ClientBuilder.build`](https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.build) returns `Result<Client>`).